knobs/file.c

```diff
@@ -8,15 +27,27 @@
 
 #include <string.h>
 
+/* if realloc fail something is particular fail 
+ * but we're going on and process the settings there fit in the memory
+ * */
+#define BREAK_REALLOC_FAIL(ptr,size) \
+    tmp = realloc(ptr,size); \
+    if (tmp == NULL) \
+        break; \
+    else \
+        ptr = tmp; \
+    tmp = NULL;
+
 int file_read_from_config(void * vp,struct config_t * cfg, char * buffer, char * prefix){
 
   int i;
   int was_set=0;
   struct file_information * info=vp;
   config_setting_t *setting;
-  /* Resetting */
+  /* Resetting Memory */
   memset(info,0,sizeof(struct file_information));
 
+  /* TODO: Find better way to parse all the file container */
```

Comment:
One way would be to allocate the memory in chunks, 10, 20, 30... Would be more efficient.